codalab/server/bundle_manager.py

```diff
@@ -695,8 +695,10 @@ def _get_matched_workers(request_queue, workers):
         """
         tag_match = re.match('tag=(.+)', request_queue)
         if tag_match != None:
-            return [worker for worker in workers if worker['tag'] == tag_match.group(1)]
-        return []
+            worker_tag = tag_match.group(1)
+            if worker_tag != None:
+                return [worker for worker in workers if worker['tag'] == worker_tag]
+        return [worker for worker in workers if worker['tag'] == request_queue]
```

Comment:
There's a simpler way to achieve this.
With regex, you can define optional, non-capturing groups so that the matched section of text doesn't change whether or not a prefix is present or not.

For this case, it would look like 
`re.match('(?:tag=)?(.+)', request_queue)`
* the `?:` in the parenthesis makes everything else inside not be captured in a match group, and the `?` right after makes it optional, so this whole regex means "if the user typed 'tag=<something>' match '<something>' otherwise match everything they wrote.

Reply:
Yes, this is the cleanest way to handle it.